### PR TITLE
Implementation of DETR using mmdetection

### PR DESCRIPTION
**Pull request by hectorLop**
This PR implements a DETR model using mmdetection. It includes:
- A new mmdet model containing the DETR config.
- Refactored the `params_group()` method so it only adds a neck layer if it exists into the model. The DETR model doesn't have any neck layer

### CODE REVIEW

**Comment by FraPochetti** [comment]
Thaks a ton for your contribution!

File: icevision/models/mmdet/models/detr/backbones/resnet_fpn.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+__all__ = [
+    "r50_8x2_150e_coco",
+]
+
+from icevision.imports import *
+from icevision.models.mmdet.utils import *
+
+
+class MMDetRetinanetBackboneConfig(MMDetBackboneConfig):
```
Comment (FraPochetti) [inline icevision/models/mmdet/models/detr/backbones/resnet_fpn.py:9]: I think it's more appropriate to rename this class to `MMDetDetrBackboneConfig`

Comment (hectorLop) [reply icevision/models/mmdet/models/detr/backbones/resnet_fpn.py]: Oh yes, I used the structure from retinanet and forgot to change the name

**Comment by FraPochetti** [comment]
Nice! all checks are passing!
I think it would be helpful to check how DETR performs on the Fridge dataset, to get some sort of benchmark, and also to make sure that everything works as expected.
Do you mind putting together a quick Colab nb with that?

You can totally use [this](https://github.com/airctic/icevision/blob/master/notebooks/getting_started_object_detection.ipynb) and install from your branch, instead of master.
This is how I installed from a remote branch of mine a while back

```
!git clone -b inference https://github.com/FraPochetti/mantisshrimp.git
%cd mantisshrimp/
!pip install -e .[all,dev]
``` 
FYI, `mantisshrimp` was the name of the repo 2 years ago :D

**Comment by hectorLop** [comment]
Yes, here is the [colab](https://colab.research.google.com/drive/1iIfZh1hACqEF3Fo0gMJwr-uvoq9GwzRQ?authuser=1#scrollTo=kzKYDkq1fEki)! I have trained the model with both fastai and pytorch lightning, but I don't know why the fastai training output disappears when reloading the notebook.

**Comment by FraPochetti** [comment]
> 

Thanks!
Can you grant View access please?
image

**Comment by hectorLop** [comment]
You should be able to view it with this [link](https://colab.research.google.com/drive/1iIfZh1hACqEF3Fo0gMJwr-uvoq9GwzRQ?usp=sharing) now.

**Comment by FraPochetti** [comment]
Great! 
Thanks a lot man.
Btw, I just tried training with a higher learning rate (`1e-3` instead of `1e-4`) as you did and managed to jump to 89% COCOMetric :)
Pretty nice.

image

**Comment by hectorLop** [comment]
Great! 
Glad to hear that the results have improved!

**Comment by ai-fast-track** [comment]
Thanks a lot @hectorLop! Awesome job!

What's your nickname on our discord. If you want you can share your Twitter and Linked handles, and I will announce your DETR integration, and tag you.

**Comment by hectorLop** [comment]
Thanks to you @ai-fast-track, I'm happy to contribute to this awesome framework!.

My nickname in discord is hlopez. These is my [LinkedIn](https://www.linkedin.com/in/hector-lopez-almazan-59b426146/) and my [Twitter](https://twitter.com/hectorLop_)

**Comment by Zubair2019** [comment]
> Yes, here is the [colab](https://colab.research.google.com/drive/1iIfZh1hACqEF3Fo0gMJwr-uvoq9GwzRQ?authuser=1#scrollTo=kzKYDkq1fEki)! I have trained the model with both fastai and pytorch lightning, but I don't know why the fastai training output disappears when reloading the notebook.

RuntimeError: Index put requires the source and destination dtypes match, got Long for the destination and Float for the source.

**Comment by hectorLop** [comment]
> > Yes, here is the [colab](https://colab.research.google.com/drive/1iIfZh1hACqEF3Fo0gMJwr-uvoq9GwzRQ?authuser=1#scrollTo=kzKYDkq1fEki)! I have trained the model with both fastai and pytorch lightning, but I don't know why the fastai training output disappears when reloading the notebook.
> 
> RuntimeError: Index put requires the source and destination dtypes match, got Long for the destination and Float for the source.

Hi @Zubair2019, could you provide more info about this error? I just executed the entire notebook, but I was unable to reproduce the error.

**Pull request merged**